test/framework/integration_framework/integration_test_framework.cpp

```diff
@@ -17,36 +17,65 @@
 
 #include "framework/integration_framework/integration_test_framework.hpp"
 
+#include <memory>
+#include "builders/protobuf/block.hpp"
+#include "builders/protobuf/proposal.hpp"
+#include "builders/protobuf/transaction.hpp"
+#include "cryptography/hash_providers/sha3_256.hpp"
+#include "datetime/time.hpp"
+#include "model/permissions.hpp"
+#include "model/transaction.hpp"
+
+using namespace shared_model::crypto;
+using namespace std::literals::string_literals;
+
 namespace integration_framework {
   IntegrationTestFramework &IntegrationTestFramework::setInitialState(
       const iroha::keypair_t &keypair) {
-    iroha::model::Block b;
-    iroha::model::generators::CommandGenerator gen;
-    iroha::model::Transaction tx;
-    for (auto v :
-         {gen.generateAddPeer("0.0.0.0:0", keypair.pubkey),
-          gen.generateCreateUserRole(default_role),
-          gen.generateCreateDomain(default_domain, default_role),
-          gen.generateCreateAccount("admin", default_domain, keypair.pubkey),
-          gen.generateCreateAsset("coin", default_domain, 1)}) {
-      tx.commands.push_back(std::move(v));
-    }
-    b.transactions = {tx};
-    b.height = 1;
-    b.txs_number = 1;
-    b.hash = iroha::hash(b);
-    return setInitialState(keypair, b);
+    Keypair key(PublicKey(keypair.pubkey.to_string()),
+                PrivateKey(keypair.privkey.to_string()));
+    auto genesis_tx =
+        shared_model::proto::TransactionBuilder()
+            .creatorAccountId("admin@test")
+            .txCounter(1)
+            .createdTime(iroha::time::now())
+            .addPeer("0.0.0.0:123", key.publicKey())
+            .createRole(
+                default_role,
+                std::vector<std::string>{iroha::model::can_create_domain,
+                                         iroha::model::can_create_account,
+                                         iroha::model::can_add_peer,
+                                         iroha::model::can_receive,
+                                         iroha::model::can_transfer})
+            .createDomain(default_domain, default_role)
+            .createAccount("admina", default_domain, key.publicKey())
+            .createAsset("coin", default_domain, 1)
+            .build()
+            .signAndAddSignature(key);
+    auto genesis_block =
+        shared_model::proto::BlockBuilder()
+            .transactions(
+                std::vector<shared_model::proto::Transaction>{genesis_tx})
```

Comment:
Please add TODO and task to allow initializer list.

Reply:
I think that TODO should be in the builders and thus it's not really good idea to add this todo here